sdks/go/examples/streaming_wordcap/wordcap.go

```diff
@@ -0,0 +1,81 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// streaming_wordcap is a toy streaming pipeline that uses PubSub. It
+// does the following:
+//    (1) create a topic and publish a few messages to it
+//    (2) start a streaming pipeline that converts the messages to
+//        upper case and logs the result.
+//
+// NOTE: it only runs on Dataflow and must be manually cancelled.
+package main
+
+import (
+	"context"
+	"flag"
+	"os"
+	"strings"
+
+	"github.com/apache/beam/sdks/go/pkg/beam"
+	"github.com/apache/beam/sdks/go/pkg/beam/core/util/stringx"
+	"github.com/apache/beam/sdks/go/pkg/beam/io/pubsubio"
+	"github.com/apache/beam/sdks/go/pkg/beam/log"
+	"github.com/apache/beam/sdks/go/pkg/beam/options/gcpopts"
+	"github.com/apache/beam/sdks/go/pkg/beam/util/pubsubx"
+	"github.com/apache/beam/sdks/go/pkg/beam/x/beamx"
+	"github.com/apache/beam/sdks/go/pkg/beam/x/debug"
+)
+
+var (
+	input = flag.String("input", os.ExpandEnv("$USER-wordcap"), "Pubsub input topic.")
+)
+
+var (
+	data = []string{
+		"foo",
+		"bar",
+		"baz",
+	}
+)
+
+func main() {
+	flag.Parse()
+	beam.Init()
+
+	ctx := context.Background()
+	project := gcpopts.GetProject(ctx)
+
+	log.Infof(ctx, "Publishing %v messages to: %v", len(data), *input)
+
+	defer pubsubx.CleanupTopic(ctx, project, *input)
+	sub, err := pubsubx.Publish(ctx, project, *input, data...)
+	if err != nil {
+		log.Fatal(ctx, err)
+	}
+
+	log.Infof(ctx, "Running streaming wordcap with subscription: %v", sub.ID())
+
+	p := beam.NewPipeline()
+	s := p.Root()
+
+	col := pubsubio.Read(s, project, *input, &pubsubio.ReadOptions{Subscription: sub.ID()})
+	str := beam.ParDo(s, stringx.FromBytes, col)
+	cap := beam.ParDo(s, strings.ToUpper, str)
```

Comment:
<!--new_thread; commit:72ebac470cd283e9389a3bee3d7fbab5b564ff70; resolved:0-->
In PR 5011, we're registering a ton of functions. Does that need to be done here?

Reply:
It does not. Function registration is just a tiny performance optimization to avoid hitting the symbol table to find that function on the worker. It has no semantic impact. I'd prefer to not do it for the examples to keep them simpler.